src/modules/helloacl.c

```diff
@@ -0,0 +1,202 @@
+/* ACL API example - An example of performing custom password authentication
```

Comment:
Please look at `tests/unit/moduleapi`  and `tests/modules` and consider adding real tests in addition to an example module. I think we now have a pretty good approach for Module API testing and it would be a good idea to maintain good test coverage for any new functionality.

Reply:
😅 I agree, if the overall idea looks good I'll follow up with tests and API docs.